unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -102,6 +147,14 @@ public async UniTask<AllFriendsInitializationMessage> GetInitializationInformati
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            if (socialClient == null) { await WaitForSocialClient(cancellationToken); }
+
+            if (socialClient == null)
+            {
+                Debug.LogError("Getting all friend requests failed due to lack of client");
+                return (new List<FriendRequest>(), new List<FriendRequest>());
```

Comment:
i would rather prefer to throw an exception than returning an invalid value so the inner layers can react accordingly

Reply:
This is OOS, will review in the future